File: TelerikUIExamples/TelerikUIExamples/OptionInfo.m
//
//  OptionInfo.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "OptionInfo.h"

@implementation OptionInfo

- (instancetype)initWithText:(NSString*)text selector:(SEL)selector
{
    return [self initWithText:text selector:selector tag:nil];
}

- (instancetype)initWithText:(NSString *)text selector:(SEL)selector tag:(id)tag
{
    self = [self init];
    if (self) {
        _tag = tag;
        _optionText = text;
        _selector = selector;
    }
    return self;
}

@end

File: TelerikUIExamplesInCSharp/Infrastructure/OptionInfo.cs
using System;

using Foundation;
using UIKit;

namespace Examples
{
	public class OptionInfo
	{
		public string OptionText { get; set; }

		public EventHandler Handler { get; set; }

		public object Tag { get; set; }

		public OptionInfo (string text, EventHandler handler)
		{
			this.OptionText = text;
			this.Handler = handler;
		}

		public OptionInfo (string text, EventHandler handler, object tag)
		{
			this.OptionText = text;
			this.Handler = handler;
			this.Tag = tag;
		}
	}
}

